app/controllers/types_controller.rb

```diff
@@ -0,0 +1,63 @@
+class TypesController < ApplicationController
+  before_action :authenticate_user!
+  load_and_authorize_resource
+  before_action :set_type, only: [:show, :edit, :update, :destroy]
+  respond_to :js, :html
+
+  def index
+    @types = current_user.types.order('created_at DESC').page(params[:page]).per(4)
```

Comment:
@VanjaJeftic please read this and modify accordingly: https://github.com/CanCanCommunity/cancancan#32-loaders and linked page https://github.com/CanCanCommunity/cancancan/blob/master/docs/Authorizing-controller-actions.md

`load_and_authorize_resource` will do it's magic and load all Types already (you can see a query in Rails log). Effectively you are doubling that work. Now, I understand that you probably want only user's Types here (I might have even asked for it) - so you either want to skip the loading of resources for index or modify it somehow to load only those you need.

Either way, the goal is to have only one database query.

Reply:
@basti I am not sure what to do, can you please help me?  First query is for pagination to return total pages param (query: SELECT COUNT(*) FROM "types" WHERE "types"."user_id" = $1), second query is connected with @types = .... from controller. 
If I put load_and_authorize_resource: except => [: index] at the beginning of the controller, I will still have the same problem.

I read all the documents that you sent to me and I understand that my index method will have @types value by default from canCan. 

To specify the problem, how to get rid of the first query because I need it for pagination, and I also need a second query for returning an adequate list of types from the controller?

Reply:
@VanjaJeftic can you please paste me the whole Rails log you see when you call this action? I need it to see what exactly is going on to be able to tell you how to fix it. I might also be wrong and that you don’t have any extra queries.

Reply:
@basti, picture of a rails log: 
![log-output](https://user-images.githubusercontent.com/57172843/102763337-ce872800-4379-11eb-89a3-e6bdbefe7dcb.png)


The fifth line from above is the result of the built-in method (part of the kaminari) which requires total_pages param and it runs automatically, if there is pagination, while the query on line 7 is the result of loading the index method from the controller.

Reply:
@VanjaJeftic ok, so you are not actually triggering a double query to fetch data (count query is necessary), but that’s only because Rails is lazy-loading data. What is happening here is that CanCanCan is defining `@types` as a query to get all data for the current user, then you ignore what’s in `@types` and you redefined it, so the query is not triggered.

My comment was aimed at using `load_and_authorize_resource` and then ignoring it. You have two possible solutions: 1) remove/skip it for index action and only use `authorize_resource` (or perform manual authorization by loading only accessible types, as you are doing currently) or 2) make it work with you action.

In case of the second solution you could write something like:

```suggestion
    @types = @types.order('created_at DESC').page(params[:page]).per(4)
```

On already defined `@types` ActiveRecord query that includes authorization based on CanCanCan you are chaining ordering and pagination.